exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/OptionIterator.java

```diff
@@ -36,7 +36,7 @@
 //  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(OptionIterator.class);
 
   enum Mode {
-    BOOT, SYS_SESS, BOTH
+    BOOT, SYS_SESS_EXTERNAL, SYS_SESS_INTERNAL, BOTH
```

Comment:
Can we improve these names? Maybe `RUNTIME` for system/session option.

I'm not clear on the distinction between external and internal. Do we mean something like public and internal?

Does the external mean only public, while internal means public and private?

And, given that we have three options, `BOTH` seems a bit of a misnomer...